### PR TITLE
Unit tests merge improvements

### PR DESCRIPTION
**Pull request by wzieba**
### Description

This PR aligns unit tests CI configuration:
- use `merge_junit_reports` script from CI toolkit
- upload only merged report file to build artifacts

### Testing information

Not needed, green CI check and demo below (or in build of [5c026df](https://github.com/woocommerce/woocommerce-android/pull/12295/commits/5c026df11d775afd10ebad55a8786e332d22fa2d) commit) should be enough.

### Images/gif

image

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: .buildkite/commands/run-unit-tests.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ echo "--- 🚦 Report Tests Status"
 results_file="WooCommerce/build/test-results/merged-test-results.xml"
 # Merge JUnit results into a single file (for performance reasons with reporting)
 # See https://github.com/woocommerce/woocommerce-android/pull/12064
-./merge_junit.sh -d WooCommerce/build/test-results/testJalapenoDebugUnitTest -o $results_file
+merge_junit_reports -d WooCommerce/build/test-results/testJalapenoDebugUnitTest -o $results_file
```
Comment (AliSoftware) [inline .buildkite/commands/run-unit-tests.sh]: Nitpick to make [`shellcheck`](https://github.com/koalaman/shellcheck) happy:
```suggestion
merge_junit_reports -d WooCommerce/build/test-results/testJalapenoDebugUnitTest -o "$results_file"
```

File: .buildkite/commands/run-unit-tests.sh [hunk 2]
```diff
@@ -31,6 +31,6 @@ echo "--- ⚒️ Generating and uploading code coverage"
 .buildkite/commands/upload-code-coverage.sh
 
 echo "--- 🧪 Copying test logs for test collector"
-mkdir WooCommerce/build/buildkite-test-analytics && cp WooCommerce/build/test-results/*.xml WooCommerce/build/buildkite-test-analytics
+mkdir WooCommerce/build/buildkite-test-analytics && cp $results_file WooCommerce/build/buildkite-test-analytics
```
Comment (AliSoftware) [inline .buildkite/commands/run-unit-tests.sh]: Ditto
```suggestion
mkdir WooCommerce/build/buildkite-test-analytics && cp "$results_file" WooCommerce/build/buildkite-test-analytics
```

**Review by AliSoftware — APPROVED** [review]
Left nitpicks to secure shell code otherwise LGTM 👍

**Pull request merged**